src/stactools/nisar_sim/stac.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import re
+
+from pystac import Collection, Item, Summaries
+from pystac.extensions.item_assets import ItemAssetsExtension
+from pystac.extensions.projection import ProjectionExtension
+from pystac.extensions.sar import SarExtension
+from pystac.extensions.sat import SatExtension
+
+from stactools.nisar_sim import constants as c
+from stactools.nisar_sim.metadata import (
+    AnnotatedMetadata,
+    HDF5Metadata,
+    Metadata,
+    MetadataLinks,
+    fill_sar_properties,
+    fill_sat_properties,
+)
+from stactools.nisar_sim.nisar_assets import get_assets
+
+# import rasterio
+
+
+logger = logging.getLogger(__name__)
+
+
+def create_collection() -> Collection:
+    """Create a STAC Collection
+
+    This function includes logic to extract all relevant metadata from
+    an asset describing the STAC collection and/or metadata coded into an
+    accompanying constants.py file.
+
+    See `Collection<https://pystac.readthedocs.io/en/latest/api.html#collection>`_.
+
+    Returns:
+        Collection: STAC Collection object
+    """
+    summary_dict = {
+        "platform": c.NISAR_SIM_PLATFORMS,
+    }
+
+    collection = Collection(
+        id="nisar-sim",
+        title="Simulated NISAR",
+        description=c.NISAR_SIM_DESCRIPTION,
+        providers=c.NISAR_SIM_PROVIDERS,
+        extent=c.NISAR_SIM_EXTENT,
+        summaries=Summaries(summary_dict),
+        stac_extensions=[
+            ItemAssetsExtension.get_schema_uri(),
+            ProjectionExtension.get_schema_uri(),
+            SarExtension.get_schema_uri(),
+            # SatExtension.get_schema_uri(),
+        ],
+        keywords=c.NISAR_SIM_KEYWORDS,
+    )
+
+    # Links
+    collection.add_links(c.NISAR_SIM_LINKS)
+
+    # SAR Extension
+    sar = SarExtension.summaries(collection, add_if_missing=True)
+    sar.looks_range = c.NISAR_SIM_SAR["looks_range"]
+    sar.product_type = c.NISAR_SIM_SAR["product_type"]
+    sar.looks_azimuth = c.NISAR_SIM_SAR["looks_azimuth"]
+    sar.polarizations = c.NISAR_SIM_SAR["polarizations"]
+    sar.frequency_band = c.NISAR_SIM_SAR["frequency_band"]
+    sar.instrument_mode = c.NISAR_SIM_SAR["instrument_mode"]
+    sar.center_frequency = c.NISAR_SIM_SAR["center_frequency"]
+    # sar.resolution_range = c.NISAR_SIM_SAR["resolution_range"]
+    # sar.resolution_azimuth = c.NISAR_SIM_SAR["resolution_azimuth"]
+    # sar.pixel_spacing_range = c.NISAR_SIM_SAR["pixel_spacing_range"]
+    # sar.pixel_spacing_azimuth = c.NISAR_SIM_SAR["pixel_spacing_azimuth"]
+    # sar.looks_equivalent_number = c.NISAR_SIM_SAR["looks_equivalent_number"]
+
+    # sat = SatExtension.summaries(collection, add_if_missing=True)
+    # sat.absolute_orbit = c.NISAR_SIM_SAT["absolute_orbit"]
+
+    assets = ItemAssetsExtension.ext(collection, add_if_missing=True)
+    assets.item_assets = get_assets(collection=True)
+
+    return collection
+
+
+def create_item(product_path: str, dither: str) -> Item:
+    """Create a STAC Item
+
+    Args:
```

Comment:
I suggest a default of X (do these letter come from the file name codes?), and then document this on the cli too.